qa/rpc-tests/test_framework/util.py

```diff
@@ -399,26 +400,30 @@ def wait_and_assert_operationid_status(node, myopid, in_status='success', in_err
     assert_true(result is not None, "timeout occured")
     status = result['status']
 
-    txid = None
+    debug = os.getenv("PYTHON_DEBUG", "")
+    if debug:
+        print('...returned status: {}'.format(status))
+
     errormsg = None
     if status == "failed":
         errormsg = result['error']['message']
-    elif status == "success":
-        txid = result['result']['txid']
-
-    if os.getenv("PYTHON_DEBUG", ""):
-        print('...returned status: {}'.format(status))
-        if errormsg is not None:
+        if debug:
             print('...returned error: {}'.format(errormsg))
-    
-    assert_equal(in_status, status, "Operation returned mismatched status. Error Message: {}".format(errormsg))
-
-    if errormsg is not None:
         assert_true(in_errormsg is not None, "No error retured. Expected: {}".format(errormsg))
```

Comment:
I don't understand this assertion message. This is the case where we *did* have an error but didn't specify an expected error message in the test, right?

```suggestion
         assert_true(in_errormsg is not None, "Unexpected error returned: {}".format(errormsg))
```

Reply:
Looking back at this, I don't really understand it either. I am changing it to check `assert_equal(in_errormsg, errormsg)` in the `if status == "failed":` block rather than the two asserts that we have.